### PR TITLE
corrected axis and weighted average

### PR DESCRIPTION
**Pull request by dimaykerby**

### CODE REVIEW

File: Training/python/DisTauTag/Training_DisTauTag_SNNv1.py [hunk 1]
```diff
@@ -18,6 +18,7 @@
 from tensorflow.keras.layers import Input, Dense, Conv2D, Dropout, AlphaDropout, Activation, BatchNormalization, Flatten, \
                                     Concatenate, PReLU, TimeDistributed, LSTM, Masking
 from tensorflow.keras.callbacks import Callback, ModelCheckpoint, CSVLogger
+from tensorflow.python.framework.convert_to_constants import convert_variables_to_constants_v2_as_graph
```
Comment (shedprog) [inline Training/python/DisTauTag/Training_DisTauTag_SNNv1.py:21]: Remove not needed modification

File: Training/python/DisTauTag/Training_DisTauTag_SNNv1.py [hunk 2]
```diff
@@ -83,19 +84,34 @@ def call(self, x, mask):
         D = batch_distance_matrix_general(coor,coor)    # (N, P, P)
         D = tf.expand_dims(D, axis=-1)  # (N, P, P, 1)
         W = tf.math.exp(-10*D)  # (N, P, P, 1)
+
+        #coor = x[:, :, -self.n_dim:]
+        #print(f"coor_after = {coor.shape}")
+        #x = x[:, :, :-self.n_dim]
```
Comment (shedprog) [inline Training/python/DisTauTag/Training_DisTauTag_SNNv1.py:90]: Remove not needed modification

File: Training/python/DisTauTag/Training_DisTauTag_SNNv1.py [hunk 3]
```diff
@@ -83,19 +84,34 @@ def call(self, x, mask):
...
         mask_dim = tf.tile(mask, (1, P ,1))    # (N, P*P, 1)
         mask_dim = tf.reshape(mask_dim, (N, P, P,1))    # (N, P, P, 1)
 
+        diagonal = tf.ones(P) #(P)
+        diag_tensor = tf.linalg.tensor_diag(diagonal) #(P,P)
+        diag_tensor = tf.expand_dims(diag_tensor, axis=-1) #(P,P,1)
+        diag_tensor = tf.tile(diag_tensor, (int(N/P), P ,1))  #(N,P*P,1)
+        diag_tensor =  tf.reshape(diag_tensor, (N, P, P,1)) #(N,P,P,1)
+        diag_tensor = -1*diag_tensor+1
+
```
Comment (shedprog) [inline Training/python/DisTauTag/Training_DisTauTag_SNNv1.py:105]: Remove not needed modification

File: Training/python/DisTauTag/Training_DisTauTag_SNNv1.py [hunk 4]
```diff
@@ -83,19 +84,34 @@ def call(self, x, mask):
...
         s = na * W * mask_dim   # (N, P, P, n_features)
-        ss = tf.math.reduce_sum(s, axis = 1)    # (N, P, n_features)
+        ss = tf.reduce_sum(s, axis=2)
+
+        #tensorrr = tf.reduce_sum(mask_dim-diag_tensor,axis=2)
+        norm_ = tf.tile(tf.reduce_sum(mask_dim*diag_tensor, axis=2),(1,1,x_shape[2]))
```
Comment (shedprog) [inline Training/python/DisTauTag/Training_DisTauTag_SNNv1.py:110]: Add the final model

File: Training/python/DisTauTag/Training_DisTauTag_SNNv1.py [hunk 5]
```diff
@@ -294,15 +314,42 @@ def main(cfg: DictConfig) -> None:
...
+        #frozen_func, graph_def = convert_variables_to_constants_v2_as_graph(real_model)
+
+        #run_meta = tf.compat.v1.RunMetadata()
+        #opts = tf.compat.v1.profiler.ProfileOptionBuilder.float_operation()
+        #flops = tf.compat.v1.profiler.profile(graph=frozen_func.graph,
+                                     #         run_meta=run_meta, cmd='op', options=opts)
+        #Flops = flops.total_float_ops
+        #print(f"Flops = {Flops}")    
+
+
```
Comment (shedprog) [inline Training/python/DisTauTag/Training_DisTauTag_SNNv1.py:341]: Remove not needed modification

File: Training/python/DisTauTag/Training_DisTauTag_SNNv1.py [hunk 6]
```diff
@@ -294,15 +314,42 @@ def main(cfg: DictConfig) -> None:
...
         mlflow.log_dict(training_cfg, 'input_cfg/training_cfg.yaml')
         mlflow.log_artifact(scaling_cfg, 'input_cfg')
         mlflow.log_artifact(to_absolute_path("Training_DisTauTag_SNNv1.py"), 'input_cfg')
         mlflow.log_artifact(to_absolute_path("../commonReco.py"), 'input_cfg')
         mlflow.log_artifacts('.hydra', 'input_cfg/hydra')
         mlflow.log_artifact('Training_DisTauTag_SNNv1.log', 'input_cfg/hydra')
         mlflow.log_param('run_id', run_id)
+        #mlflow.log_param('FLOPs', flops)
+        #mlflow.log_param('trainable_params', trainable_count)
```
Comment (shedprog) [inline Training/python/DisTauTag/Training_DisTauTag_SNNv1.py:350]: Remove not needed modification

**Review by shedprog — CHANGES_REQUESTED** [review]
